components/sup/src/ctl_gateway/server.rs

```diff
@@ -18,7 +18,6 @@ use futures::{channel::mpsc,
               ready,
               task::{Context,
                      Poll}};
-use habitat_common::liveliness_checker;
```

Comment:
This makes sense, as this is moving into the main multi-threaded reactor. It no longer holds to the assumption the liveness checker is based on, which is that the code in question is always executing in the same thread.

Reply:
Slightly related. I wanted to use our liveness checker in the threads spawned by tokio. I was close to getting it to work using the [`on_thread_start`](https://docs.rs/tokio/0.2.4/tokio/runtime/struct.Builder.html#method.on_thread_start) and [`on_thread_stop`](https://docs.rs/tokio/0.2.4/tokio/runtime/struct.Builder.html#method.on_thread_stop) methods of the `Builder`. I wanted to spawn a never ending task that would periodically call `mark_thread_alive` using [`LocalSet`](https://docs.rs/tokio/0.2.4/tokio/task/struct.LocalSet.html). Unfortunately, the way `LocalSet` works you can only spawn task by blocking the runtime so this ultimately did not work. Maybe it will be possible one day?